### DISCUSSION TITLE
is_accessible (jwt-token)

### DISCUSSION THREAD

**Discussion by wnesbv** [aminalaee/sqladmin#98] [Q&A] [answered]
`
    For me, the whole authorization authentication boils down to this:

    oauth2_scheme = OAuth2PasswordBearerWithCookie(tokenUrl = "/")

    def get_user(
        db: Session = Depends(get_db),
        token: str = Depends(oauth2_scheme),
    ):

        email = auth.decode_token(token)
        user = db.query(User).filter(User.email == email).first()
        if not user:
            raise HTTPException(401, "User doesn't exist")

        return user

    def get_admin(
        current_user: UserOut = Depends(get_user)
    ):
        if current_user.is_admin:
            return current_user

    def get_active_user(
        current_user: UserOut = Depends(get_user)
    ):
        if not current_user.email_verified:
            raise HTTPException(401, "Email not confirmed..!")
        if not current_user.is_active:
            raise HTTPException(401, "This account is not active..!")

        return current_user

    But, I can't use it here.

    class AuthModelAdmin(ModelAdmin):
        def is_accessible(
            self,
            request: Request,
        ) -> bool:

            return True

    And this is my first self-authorization-authentication. Maybe there are solutions..?
`

**Comment by aminalaee** [comment]
Hey,
So if I understand the code correctly, you use `get_user ` to query the user by Token?

Then maybe something like this?

```python
class AuthModelAdmin(ModelAdmin):
    def is_accessible(
        self,
        request: Request,
    ) -> bool:

        token = request.headers.get("Authentication")
        # Calling whatever function to handle the auth
         user = get_user(db, token)
         if not user.admin:
                return False
         return True
```

Does that make sense?

**Reply by wnesbv** [reply]
`
        in my case,. rather so..

        token = request.cookies.get("access_token")
        user = get_user(token)
        if not user.admin:
            return False
        return True
`

**Reply by wnesbv** [reply]
401
Invalid token

**Reply by aminalaee** [reply]
Can you check the token to see if it has any values?

**Reply by wnesbv** [reply]
{
  "exp": 1647857588,
  "iat": 1647856688,
  "scope": "access_token",
  "sub": "admin@gmail.com"
}

**Reply by wnesbv** [reply]
"Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2NDc4NTc1ODgsImlhdCI6MTY0Nzg1NjY4OCwic2NvcGUiOiJhY2Nlc3NfdG9rZW4iLCJzdWIiOiJhZG1pbkBnbWFpbC5jb20ifQ.XSAr8WqmBoGJnQkLThgwFt3rElojXTSp1boAXe7oPK0"

**Reply by wnesbv** [reply]
I, can I pour a pack on git..?

**Reply by aminalaee** [reply]
Sorry I don't know what that means.

**Reply by wnesbv** [reply]
(Can you check the token to see if it has any values?)
{
"exp": 1647857588,
"iat": 1647856688,
"scope": "access_token",
"sub": "[admin@gmail.com](mailto:admin@gmail.com)"
}

**Reply by aminalaee** [reply]
I mean if the request and headers and cookies are present in the `is_accessible` method then it should be an internal problem, does that make sense? Or can you provide a minimal example maybe as a project or gist to take a look into it?

**Reply by wnesbv** [reply]
https://github.com/wnesbv/fastapi-web-html

**Reply by wnesbv** [reply]
add sqladmin.zip

**Comment by wnesbv** [comment, marked as answer]
I've decided everything. Two offline authorizations

**Reply by aminalaee** [reply]
Sorry, what do you mean?
